src/Silverback.Integration.MQTT/Messaging/Broker/Mqtt/MqttClientWrapper.cs

```diff
@@ -32,6 +32,8 @@ internal sealed class MqttClientWrapper : IDisposable
 
         private CancellationTokenSource? _connectCancellationTokenSource;
 
+        private bool _disconnectingCallbackCalled;
```

Comment:
I think this is necessary because `DisconnectAsync` can be called multiple times and we want the callback to be executed only once.

Reply:
The DisconnectAsync method already handle duplicate calls.

```
if (_connectedObjects.Count > 0 || !MqttClient.IsConnected)	
    return Task.CompletedTask;
```

Don't you think that's enough?